core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -686,19 +686,6 @@ protected Scope visitRefreshMaterializedView(RefreshMaterializedView refreshMate
                     .map(insertColumn -> tableMetadata.getColumn(insertColumn).getType())
                     .collect(toImmutableList());
 
-            List<Type> queryTypes = queryScope.getRelationType().getVisibleFields().stream()
```

Comment:
lack of test

Reply:
this is test covered by added REFRESH in `io.trino.testing.BaseConnectorTest#testMaterializedViewAllTypes`

Reply:
Engine only test level would be preferred, see https://github.com/trinodb/trino/commit/12fb514410b4a550dba0b3fe322623e27dc531f5 how I've added engine-only test for MV select with mismatched column types (`TestMaterializedViews`)

Reply:
Thanks for the pointer. Added `io.trino.sql.planner.TestMaterializedViews#testRefreshMaterializedViewWithCasts`